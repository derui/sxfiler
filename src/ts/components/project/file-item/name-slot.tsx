import { h } from "preact";

interface Prop {
  isDirectory: boolean;
  isSymlink: boolean;
  name: string;
}

export const Component: preact.FunctionComponent<Prop> = ({ isDirectory, name, isSymlink }) => {
  return (
    <span
      class="file-item__item-name"
      data-testid="fileItem-nameSlot"
      data-directory={isDirectory && !isSymlink}
      data-symlink={isSymlink}
    >
      {name}
    </span>
  );
};
