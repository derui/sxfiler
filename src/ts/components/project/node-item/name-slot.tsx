import classNames from "classnames";
import * as React from "react";

// tslint:disable-next-line
const styles = require("./node-item.module.scss");

interface Prop {
  isDirectory: boolean;
  isSymlink: boolean;
  name: string;
}

const NodeName: React.FC<Prop> = ({ isDirectory, name, isSymlink }) => {
  const className = classNames(styles.name, {
    [styles.nameDirectory]: isDirectory && !isSymlink,
    [styles.nameSymlink]: isSymlink,
  });

  return <span className={className}>{name}</span>;
};

export default NodeName;
