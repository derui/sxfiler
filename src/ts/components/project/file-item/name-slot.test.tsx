import { h } from "preact";
import { render } from "preact-render-to-string";
import { Component as T } from "./name-slot";

describe("Project", () => {
  describe("Node Item", () => {
    describe("Name slot", () => {
      it("should print name simply", () => {
        const tree = render(<T name="name" isDirectory={false} isSymlink={false} />);

        expect(tree).toMatchSnapshot();
      });

      it("should print name as directory", () => {
        const tree = render(<T name="name" isDirectory={true} isSymlink={false} />);

        expect(tree).toMatchSnapshot();
      });

      it("should print name as symlink", () => {
        const tree = render(<T name="name" isDirectory={false} isSymlink={true} />);

        expect(tree).toMatchSnapshot();
      });

      it("should print name as symlink when directory and symlink flag are enabled", () => {
        const tree = render(<T name="name" isDirectory={true} isSymlink={true} />);

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
