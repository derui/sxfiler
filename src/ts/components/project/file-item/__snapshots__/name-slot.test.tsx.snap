// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Node Item Name slot should print name as directory 1`] = `"<span class=\\"file-item__item-name\\" data-testid=\\"fileItem-nameSlot\\" data-directory>name</span>"`;

exports[`Project Node Item Name slot should print name as symlink 1`] = `"<span class=\\"file-item__item-name\\" data-testid=\\"fileItem-nameSlot\\" data-symlink>name</span>"`;

exports[`Project Node Item Name slot should print name as symlink when directory and symlink flag are enabled 1`] = `"<span class=\\"file-item__item-name\\" data-testid=\\"fileItem-nameSlot\\" data-symlink>name</span>"`;

exports[`Project Node Item Name slot should print name simply 1`] = `"<span class=\\"file-item__item-name\\" data-testid=\\"fileItem-nameSlot\\">name</span>"`;
